homeassistant/components/miele/sensor.py

```diff
@@ -181,10 +181,10 @@ class MieleSensorDefinition:
         description=MieleSensorDescription(
             key="state_program_type",
             translation_key="program_type",
-            value_fn=lambda value: value.state_program_type,
+            value_fn=lambda value: StateProgramType(value.state_program_type).name,
             entity_category=EntityCategory.DIAGNOSTIC,
             device_class=SensorDeviceClass.ENUM,
-            options=sorted(set(STATE_PROGRAM_TYPE.values())),
+            options=sorted(set(StateProgramType.keys())),
```

Comment:
```suggestion
            options=sorted(StateProgramType.keys()),
```

Reply:
You are right with current content in StateProgramType, when there are no duplicate values. In next PR for StateProgramPhase, there will absolutely be members with duplicate values. This will need a customised \_\_new\_\_() in MieleEnum and set() in the options arguments. I can remove `set` here and now but I will have to put it back when we discover and add a member with multiple values.

Reply:
Okay lets not change it then.